qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDefinition.java

```diff
@@ -270,6 +273,37 @@ private CompletableFuture<List<? extends LocationLink>> findDefinitionFromPart(P
 		}
 	}
 
+	/**
+	 * Find parameter definition for Enum in #when section.
+	 *
+	 * @param parameter
+	 * @param template
+	 * @param cancelChecker
+	 *
+	 * @return parameter definition for Enum in #when section.
+	 */
+	private CompletableFuture<List<? extends LocationLink>> findDefinitionFromParameter(Parameter parameter,
+			Template template, CancelChecker cancelChecker) {
+		String projectUri = template.getProjectUri();
+		Section whenSection = parameter.getParentSection().getParentSection();

```

Comment:
I think the first thing that you should do is to get the ownser section of the parameter znd check if it is a case section.
Indeed according your code if you have à #let instead of #case it will find the définition although it should not.
Please write a test with that too.